ddtrace/sampler.py

```diff
@@ -202,17 +203,34 @@ def __init__(
         # Configure rate limiter
         self.limiter = RateLimiter(rate_limit)
 
+        self._legacy_sampler = None  # type: Optional[RateByServiceSampler]
+        self._default_rule = None  # type: Optional[SamplingRule]
         if default_sample_rate is None:
             log.debug("initialized DatadogSampler, limit %r traces per second", rate_limit)
             # Default to previous default behavior of RateByServiceSampler
-            self.default_sampler = RateByServiceSampler()  # type: BaseSampler
+            self._legacy_sampler = RateByServiceSampler()
         else:
             log.debug(
                 "initialized DatadogSampler, sample %s%% traces, limit %r traces per second",
                 100 * default_sample_rate,
                 rate_limit,
             )
-            self.default_sampler = SamplingRule(sample_rate=default_sample_rate)
+            self._default_rule = SamplingRule(sample_rate=default_sample_rate)
+
+        # This shouldn't ever happen, but let's just be 100% sure
+        if not self._default_rule and not self._legacy_sampler:
+            raise ValueError("DatadogSampler configured without a default rule or fallback sampler: {!r}".format(self))
+
+    @removed_property
```

Comment:
Should we provide/customize the message on the warning to indicate that it will be removed in 1.0?